### DISCUSSION TITLE
Newbie: How do I attach a disk to a bucket?

### DISCUSSION THREAD

**Discussion by mskovdk** [minio/minio#18291] [Q&A] [answered]
Homelab setup. 
Want to Create a S3 target for my friend to backup data to.
His bucket data should physically be on a particular USB drive.
Ubuntu with docker and portainer.
Minio is installed as a Docker  container.
Host Ubuntu can see the USB partition as /dev/sdb1.

Now how to link that drive to mimio and the bucket?

Forgive me asking. I have a feeling it's basic Linux stuff. I'm at the learning stage...

**Comment by harshavardhana** [comment, marked as answer]
### Steps

- Format the drive first with XFS
  - mkfs.xfs /dev/sdb1

- Mount the drive and provide permissions for MinIO to write to this drive
  - mount /dev/sdb1 /media/content
  - chown -R minio-user. /media/content
  - minio server /media/content 

And then login via Console UI and upload files

**Reply by mskovdk** [reply]
I got as far as being able to see /dev/UsbSamsung500 (my test drive) in the console of MinIO.
(I used this mapping in stead of /media/content, recommended by another guide for simplicity)

But I'm missing something. The link from the drive to the bucket.
How do I tell the bucket that it's data must live on /dev/UsbSamsung500 ?

**Reply by mskovdk** [reply]
minio server /dev/UsbSamsung500 
ERROR Unable to start the server: Specified port is already in use

uhh... Is it one storage = one server instance? 
So if Pete and Paul need access to each a 500G drive, a personal drive, I must run two MinIO instances?
Not what I expected, but I can work with that, I guess.

**Reply by harshavardhana** [reply]
You run one instance and both of you can use the instance.

**Reply by mskovdk** [reply]
Solved it. 
Portainer and a manual creation of MinIO gave me a corrupt setup somehow. Could not change the Data path.
I started over, finding a Portainer template for MinIO. I could bind the host drive as I needed, and now it works.

I'm still a bit puzzled about the entire storage setup under MinIO, being used to Windows (sorry). I wonder if I can mount a disk under a bucket? Originally I would like to dedicate a physical host disk to a bucket. I can live without, but... 
As it is now, all future buckets would live on the same physical drive.

**Reply by klauspost** [reply]
Mount point is for all bucket, not singular buckets.

You can fake it with NTFS and [mount drives as a folder](https://learn.microsoft.com/en-us/windows-server/storage/disk-management/assign-a-mount-point-folder-path-to-a-drive), but you are on your own, since you are outside anything we'd recommend.
